### PR TITLE
feat: Button to transfer user to mobile App

### LINKED ISSUE #15091: Transfer a session to mobile app (UI button + new API endpoint)
**Is your feature request related to a problem? Please describe.**
A user can join a BBB session from many places (email link, greenlight, moodle, others).
We need a consistent way that user can be transferred for the mobile application, for instance, to be able to share the screen when using IOS.

**Describe the solution you'd like**
I want to add a new option in this menu, that says "📱 Mobile application":
image

**Describe alternatives you've considered**
We considered using universal links, but as BigBlueButton can be hosted in any server, this is not an option.

**Additional context**
In order to get this working, we will need to create a new endpoint in api (`/rejoin`), that will generate a new join URL.
This API will return a regular join URL, with an extra security parameter to mitigate session theft in case of a XSS attack. 
This parameter will be the user IP address, that will be included in the checksum, and will be checked by bbb-web when this address is called, ensuring that it's not someone else calling the URL.

This issue can be composed as:
- API change - create the `/rejoin` endpoint;
- API change - modify the `/join` endpoint to validate IP address (only when provided);
- UI change (add the button on menu);

### PR DESCRIPTION
**Pull request by gustavotrott**
- A new option was added when user device is iOS:
image

- The user will choose whether wants to open the BBB App (using a URL Scheme to achieve it) or if they want to open the App Store and download the App at this moment.
image

- Once the user opens the App and connect using the same externalId, the original user will be warned about this:
image

https://user-images.githubusercontent.com/5660191/176890412-c9b8e419-bd65-45a8-a52c-5ef8a86e2473.MP4

Closes #15091

### CODE REVIEW

**Comment by gustavotrott** [comment]
Pending: Hide "Open in the app" button when user is already in the app.
It will be done when [15295](https://github.com/bigbluebutton/bigbluebutton/pull/15295) be merged.

**Comment by lfzawacki** [comment]
Maybe we should add a setting.yml option to enable/disable this button or just skip showing it if the app url is empty.

**Comment by gustavotrott** [comment]
> Maybe we should add a setting.yml option to enable/disable this button or just skip showing it if the app url is empty.

Good idea. Now the menu option will be displayed only if `iosAppStoreUrl` is not empty.

**Comment by gustavotrott** [comment]
Hey @lfzawacki, are you able to take look when you have the chance?

File: bigbluebutton-html5/public/locales/en.json [hunk 1]
```diff
@@ -373,6 +373,7 @@
     "app.navBar.settingsDropdown.hotkeysLabel": "Keyboard shortcuts",
     "app.navBar.settingsDropdown.hotkeysDesc": "Listing of available keyboard shortcuts",
     "app.navBar.settingsDropdown.helpLabel": "Help",
+    "app.navBar.settingsDropdown.openAppLabel": "Open in the BigBlueButton app",
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/public/locales/en.json:376]: Open in BigBlueButton Tablet app

File: bigbluebutton-html5/private/config/settings.yml [hunk 2]
```diff
@@ -33,6 +33,8 @@ public:
     copyright: '©2022 BigBlueButton Inc.'
     html5ClientBuild: HTML5_CLIENT_VERSION
     helpLink: https://bigbluebutton.org/html5/
+    iosAppStoreUrl: ''
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/private/config/settings.yml:36]: Change to this value: https://apps.apple.com/us/app/bigbluebutton-tablet/id1641156756

File: bigbluebutton-html5/private/config/settings.yml [hunk 3]
```diff
@@ -33,6 +33,8 @@ public:
...
+    iosAppUrlScheme: bigbluebutton
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/private/config/settings.yml:37]: change to this scheme: bigbluebutton-tablet

File: bigbluebutton-html5/public/locales/en.json [hunk 4]
```diff
@@ -400,6 +401,15 @@
     "app.about.confirmDesc": "OK",
     "app.about.dismissLabel": "Cancel",
     "app.about.dismissDesc": "Close about client information",
+    "app.mobileAppModal.title": "Open BigBlueButton app",
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/public/locales/en.json:404]: Open BigBlueButton tablet app

File: bigbluebutton-html5/public/locales/en.json [hunk 5]
```diff
@@ -400,6 +401,15 @@
...
+    "app.mobileAppModal.description": "Do you have the BigBlueButton app installed on your device?",
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/public/locales/en.json:405]: Do you have the BigBlueButton Tablet app installed on your device?

**Review by TiagoJacobs — CHANGES_REQUESTED** [review]

File: bigbluebutton-html5/imports/ui/components/nav-bar/settings-dropdown/component.jsx [hunk 6]
```diff
@@ -120,7 +126,8 @@ const defaultProps = {
 };
 
 const ALLOW_FULLSCREEN = Meteor.settings.public.app.allowFullscreen;
-const { isSafari } = browserInfo;
+const APP_CONFIG = Meteor.settings.public.app;
+const { isSafari, isMobileApp } = browserInfo;
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/imports/ui/components/nav-bar/settings-dropdown/component.jsx:130]: Please rename this variable to be called `isTabletApp` to match the new app name

File: bigbluebutton-html5/public/locales/en.json [hunk 7]
```diff
@@ -410,6 +411,15 @@
     "app.about.confirmDesc": "OK",
     "app.about.dismissLabel": "Cancel",
     "app.about.dismissDesc": "Close about client information",
+    "app.mobileAppModal.title": "Open BigBlueButton tablet app",
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/public/locales/en.json:414]:
```suggestion
    "app.mobileAppModal.title": "Open BigBlueButton Tablet app",
```

File: bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy [hunk 8]
```diff
@@ -1114,6 +1115,103 @@ class ApiController {
...
+        logData.put("description", "Request join URL.");
+        Gson gson = new Gson();
+        String logStr = gson.toJson(logData);
+
+        log.info(" --analytics-- data=" + logStr);
+
+        String method = 'join'
+        String extId = validationService.encodeString(meeting.getExternalId())
+        String fullName = validationService.encodeString(us.fullname)
+        String query = "fullName=${fullName}&meetingID=${extId}&password=${us.role.equals(ROLE_MODERATOR) ? meeting.moderatorPassword : meeting.viewerPassword}&redirect=true&userID=${us.getExternUserID()}"
```
Comment (TiagoJacobs) [inline bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy:1196]: `password` parameter is deprecated in API, please use the `role` parameter instead.

**Review by TiagoJacobs — CHANGES_REQUESTED** [review]

**Pull request merged**